docker_cli/image.py

```
@@ -0,0 +1,430 @@
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile_string : str
+            A Dockerfile-formatted string containing the information necessary
+            to build the desired image.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
+            program will be written to. If None, the method will not output to
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile-formatted string at context/dockerfile_loc
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        context,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        output_file=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None:
+            dockerfile_build = True
+        elif dockerfile_string is not None:
+            dockerfile_build = False
+        else:
+            return ValueError("Image.build: either dockerfile or " +
+                              "dockerfile_string must be defined.")
```

Let's add an initial check that the user didn't accidentally provide \*both\* `dockerfile` \*and\* `dockerfile_string` (i.e. we should check whether both were not None). We should raise a `ValueError` in that case since we won't know whether to use the dockerfile or the string.

Currently, in that case we just use `dockerfile` and silently ignore `dockerfile_string`, which might not be what the user expected.